massa-execution-worker/src/execution.rs

```diff
@@ -901,7 +828,12 @@ impl ExecutionState {
             .active_cursor
             .get_next_slot(self.config.thread_count)
             .expect("slot overflow when getting speculative ledger");
-        let search_result = self.fetch_active_history_data_entry(next_slot, address, key);
+        self.verify_active_slot(next_slot);
+        let index = self.get_active_index(next_slot);
```

Comment:
here as well, the index is None because it's beyond the history end

Reply:
See https://github.com/massalabs/massa/pull/2701#discussion_r901531034